### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by paulcpederson**
I broke a couple tests from the test suite with my recent prs. This gets everything but the card ones working again.

### CODE REVIEW

File: src/components/calcite-icon/utils.ts [hunk 1]
```diff
@@ -35,21 +35,24 @@ export async function fetchIcon({
   const size = scaleToPx[scale];
   const id = `${normalizeIconName(icon)}${size}${filled ? "F" : ""}`;
 
-  return new Promise(resolve => {
-    if (requestCache[id]) {
-      return resolve(requestCache[id]);
+  return new Promise(async function(resolve) {
```
Comment (jcfranco) [inline src/components/calcite-icon/utils.ts]: This top-level Promise isn't needed since the function is already `async`.

Comment (paulcpederson) [reply src/components/calcite-icon/utils.ts]: @jcfranco can you unpack that a little. `async` is completely new to me and I'd appreciate the pointers

**Comment by paulcpederson** [comment]
@jcfranco thanks, that was super helpful!

File: src/components/calcite-icon/utils.ts (hunk 1 above)
Comment (jcfranco) [reply src/components/calcite-icon/utils.ts]: Sure. Basically when you use the `async` keyword on a function, the browser (or a generator helper if transpiling to ES5) will wrap it in a promise. 🤓

File: src/demos/calcite-icon.html [hunk 2]
```diff
@@ -60,8 +55,8 @@ <h2>Dark theme</h2>
   <h2>Flips icon in RTL when mirrored</h2>
   <div dir="rtl">
     <calcite-icon icon="aZ" filled mirror></calcite-icon>
```
Comment (jcfranco) [inline src/demos/calcite-icon.html]: One more leftover `filled` attribute here.

**Pull request merged**